File: check.php
<?php
// ตรวจสอบประเภทของเบราว์เซอร์จาก User-Agent
$userAgent = $_SERVER['HTTP_USER_AGENT'];
$browser = '';

if (strpos($userAgent, 'MSIE') !== false) {
    $browser = 'MSIE';
} elseif (strpos($userAgent, 'Trident') !== false) {
    $browser = 'Trident';
} elseif (strpos($userAgent, 'Firefox') !== false) {
    $browser = 'Firefox';
} elseif (strpos($userAgent, 'Chrome') !== false) {
    $browser = 'Chrome';
} elseif (strpos($userAgent, 'Opera Mini') !== false) {
    $browser = 'Opera Mini';
} elseif (strpos($userAgent, 'Opera') !== false) {
    $browser = 'Opera';
} elseif (strpos($userAgent, 'Safari') !== false) {
    $browser = 'Safari';
} elseif (strpos($userAgent, 'Mozilla') !== false) {
    $browser = 'Mozilla';
}

echo "\n" . $browser;

// บันทึกข้อมูลผู้ใช้งานลงในไฟล์ logs.txt
$protocol = $_SERVER['SERVER_PROTOCOL'];
$ip = $_SERVER['REMOTE_ADDR'];
$port = $_SERVER['REMOTE_PORT'];
$hostname = gethostbyaddr($ip);

$logData = "IP: $ip\nProtocol: $protocol\nPort: $port\nUser-Agent: $userAgent\nHostname: $hostname\n\n";
file_put_contents('logs.txt', $logData, FILE_APPEND);

// ตรวจสอบคีย์ใน Whitelist
$keys = ["Key-12", "key-34"];
$sub = $_GET["key"] ?? '';

echo in_array($sub, $keys, true) ? "Whitelisted" : "Not Whitelisted";
?>

File: check.lua
-- ฟังก์ชันเพื่อตรวจสอบประเภทของเบราว์เซอร์จาก User-Agent
local userAgent = os.getenv("HTTP_USER_AGENT")
local browser = ""

if userAgent:find("MSIE") then
    browser = "MSIE"
elseif userAgent:find("Trident") then
    browser = "Trident"
elseif userAgent:find("Firefox") then
    browser = "Firefox"
elseif userAgent:find("Chrome") then
    browser = "Chrome"
elseif userAgent:find("Opera Mini") then
    browser = "Opera Mini"
elseif userAgent:find("Opera") then
    browser = "Opera"
elseif userAgent:find("Safari") then
    browser = "Safari"
elseif userAgent:find("Mozilla") then
    browser = "Mozilla"
end

print("\n" .. browser)

-- ฟังก์ชันบันทึกข้อมูลผู้ใช้งานลงในไฟล์ logs.txt
local protocol = os.getenv("SERVER_PROTOCOL")
local ip = os.getenv("REMOTE_ADDR")
local port = os.getenv("REMOTE_PORT")
local hostname = io.popen("nslookup " .. ip):read("*a"):match("name = ([%w%.%-]+)")

local logData = string.format("IP: %s\nProtocol: %s\nPort: %s\nUser-Agent: %s\nHostname: %s\n\n", 
    ip, protocol, port, userAgent, hostname or "N/A")

local file = io.open("logs.txt", "a")
file:write(logData)
file:close()

-- ฟังก์ชันตรวจสอบคีย์ใน Whitelist
local keys = { "Key-12", "key-34" }
local sub = os.getenv("QUERY_STRING"):match("key=([^&]*)") or ""

local whitelisted = false
for _, key in ipairs(keys) do
    if key == sub then
        whitelisted = true
        break
    end
end

print(whitelisted and "Whitelisted" or "Not Whitelisted")
